src/AprsService.scala

```diff
@@ -230,18 +231,89 @@ class AprsService extends Service {
 			Digipeater.parseList(digipath, true), payload)
 	}
 
+	def newPacketMice(payload: String, destString: String) = {
+		val digipath = prefs.getString("digi_path", "WIDE1-1")
+		val parsedDigipath = Digipeater.parseList(digipath, true)
+		val callsign_ssid = prefs.getCallSsid()
+		Log.d("newPacketMice", s"digipath retrieved: $digipath")
+		
+		// Construct the micePacket string
+		val micePacketString = s"$callsign_ssid>$destString${if (digipath.nonEmpty) s",$digipath" else ""}:$payload"
+		
+		// Log or return the constructed packet
+		Log.d("newPacketMice", s"Constructed MICE Packet: $micePacketString")
+
+		val micePacketParsed = Parser.parse(micePacketString)
+		
+		micePacketParsed
+	}
 	def formatLoc(symbol : String, status : String, location : Location) = {
```

Comment:
[optional] not sure if it would look cleaner if we'd split the compressed packet encoding into `formatLocCompressed` and then call that separately, instead of treating both cases from one method.